xbmc/utils/log.cpp

```diff
@@ -33,7 +33,51 @@ namespace
 constexpr unsigned char Utf8Bom[3] = {0xEF, 0xBB, 0xBF};
 const std::string LogFileExtension = ".log";
 const std::string LogPattern = "%Y-%m-%d %T.%e T:%-5t %7l <%n>: %v";
-} // namespace
+
+struct ComponentInfo
+{
+  std::string name;
+  uint32_t stringId{0};
+};
+
+// clang-format off
+const std::map<int, ComponentInfo> componentMap = {
```

Comment:
Can we use a `constexpr CMap` here so the calls are translated at compile time?

Reply:
@sundermann where in our code base can I find `CMap` use cases?

Reply:
There's only a few but this could be a good place for another usage since the logging function is called often so we'd save on the O(log(n)) additional lookup.